src/trace.rs

```diff
@@ -47,10 +47,17 @@ pub fn init(color: bool, json: bool, levels: &str) {
     // bit of duplication as we started to create a generic struct to wrap the formatters that also
     // implemented `Layer`
     let dispatch = if json {
+        #[cfg(not(test))]
         let formatter = tracing_subscriber::fmt::Layer::default()
             .json()
             .flatten_event(true);
 
+        #[cfg(test)]
+        let formatter = tracing_subscriber::fmt::Layer::default()
+            .json()
+            .flatten_event(true)
+            .with_test_writer(); // ensures output is captured
+
```

Comment:
One way to tidy this up might be to use the `cfg!` macro. Then you'd have 

```rust
if cfg!(test) {
   ...
} else {
   ...
}
```

That said, I'm happy with the current setup if you like it. A thought.

Reply:
I actually tried this first, but I couldn't figure out what the type of `formatter` should be to unify both arms.

Reply:
This is the error you would see first:

```
error[E0308]: `if` and `else` have incompatible types
  --> src/trace.rs:48:9
   |
45 |       let subscriber = if cfg!(test) {
   |  ______________________-
46 | |         subscriber.with(tracing_subscriber::fmt::layer().with_test_writer())
   | |         -------------------------------------------------------------------- expected because of this
47 | |     } else {
48 | |         subscriber
   | |         ^^^^^^^^^^ expected struct `tracing_subscriber::fmt::Layer`, found struct `EnvFilter`
49 | |     };
   | |_____- `if` and `else` have incompatible types
   |
   = note: expected type `Layered<tracing_subscriber::fmt::Layer<Layered<EnvFilter, tracing_subscriber::Registry>, DefaultFields, tracing_subscriber::fmt::format::Format, TestWriter>, Layered<EnvFilter,
 tracing_subscriber::Registry>, Layered<EnvFilter, tracing_subscriber::Registry>>`
            found struct `Layered<EnvFilter, tracing_subscriber::Registry, tracing_subscriber::Registry>`
```